pkg/controller/job/job_controller.go

```diff
@@ -424,34 +457,37 @@ func (jm *JobController) syncJob(key string) error {
 		now := metav1.Now()
 		job.Status.StartTime = &now
 	}
-	// if job was finished previously, we don't want to redo the termination
-	if IsJobFinished(&job) {
-		return nil
-	}
 
+	jobFailed := false
 	var manageJobErr error
-	if pastActiveDeadline(&job) {
+	var newFailureConditionFunc func(record.EventRecorder) batch.JobCondition
+
+	jobHaveNewFailure := failed > job.Status.Failed
+
+	// check if the number of failed jobs increaded since the last syncJob
+	if jobHaveNewFailure && jobToManyFailed(previousRetry+1, job.Spec.BackoffLimit) {
+		jobFailed = true
+		newFailureConditionFunc = func(r record.EventRecorder) batch.JobCondition {
+			r.Event(&job, v1.EventTypeNormal, "BackoffLimitExceeded", "Job has reach the specified backoff limit")
```

Comment:
Should be `EventTypeWarning` not `EventTypeNormal`.

Reply:
Don't create methods like this, just record event and save condition in tmp variable.

Reply:
I created 2 new string vars for "reason" and "message" used in by ```Event()```and ```NewCondition()```